test/metabase/query_processor_test/explicit_joins_test.clj

```diff
@@ -324,6 +326,79 @@
+  (mt/test-drivers (mt/normal-drivers-with-feature :left-join)
+    (testing "join against multiple copies of a card (#34227)"
+      (mt/dataset sample-dataset
+        (let [metadata-provider (qp.test-util/metadata-provider-with-cards-with-metadata-for-queries
+                                 [(mt/mbql-query orders
+                                    {:breakout [$user_id]
+                                     :aggregation [[:count]]})
+                                  (mt/mbql-query orders
+                                    {:breakout [$user_id]
+                                     :aggregation [[:count]]})
+                                  (mt/mbql-query people
+                                    {:fields [$id]
+                                     :joins [{:fields :all
+                                              :alias "ord1"
+                                              :source-table "card__1"
+                                              :condition [:= $id &ord1.orders.user_id]}
+                                             {:fields :all
+                                              :alias "ord2"
+                                              :source-table "card__2"
+                                              :condition [:= $id &ord2.orders.user_id]}]})])]
+          (qp.store/with-metadata-provider metadata-provider
+            (let [top-card-query (mt/mbql-query people
+                                   {:source-table "card__3"
+                                    :limit        3})
+                  [cid cuser-id ccount cuser-id2 ccount2] (->> top-card-query
+                                                               qp/query->expected-cols
+                                                               (map :name))
+                  cid2 (str cid "_2")
+                  col-data-fn   (juxt            :id       :name     :source_alias)
+                  top-card-cols [[(mt/id :people :id)      cid       nil]
+                                 [(mt/id :orders :user_id) cuser-id  "ord1"]
+                                 [nil                      ccount    "ord1"]
+                                 [(mt/id :orders :user_id) cuser-id2 "ord2"]
+                                 [nil                      ccount2   "ord2"]]]
+              (testing "sanity"
+                (is (= top-card-cols
+                       (->> top-card-query
+                            qp/process-query
+                            mt/cols
+                            (map col-data-fn)))))
+
+              (testing "suggested join condition references the FK by name"
+                (let [query (lib/query metadata-provider (lib.metadata/table metadata-provider (mt/id :people)))
+                      card-meta (lib.metadata/card metadata-provider 3)]
+                  (is (=? [:= {} [:field {} (mt/id :people :id)] [:field {} cuser-id]]
+                          (lib/suggested-join-condition query card-meta)))))
```

Comment:
How bad do we feel about testing `suggested-join-condition` here? The only reason I added it here because I didn't feel like reproducing the query one more time as an MLv2 test.